### PR TITLE
Refactor masthead to use memo and fix styling

### PR DESCRIPTION
**Pull request by PatoBeltran**
Masthead now uses React.memo and the css is cleaned up to remove unnecessary rules.

### CODE REVIEW

File: lib/components/Masthead/Masthead.module.scss [hunk 1]
```diff
@@ -96,66 +26,68 @@ $search-max-size: 100 * $grid-size;
...
-      .masthead-toolbar-btn-container:not(.selected-more) {
-        @media (max-width: $screen-sm - 1) {
-          display: none;
-        }
+    .more-button {
+      @media (min-width: $screen-sm) {
+        display: none;
       }
-
+
```
Comment (mryanlo) [inline lib/components/Masthead/Masthead.module.scss]: nit: remove extra spacing

File: lib/components/Masthead/Masthead.tsx [hunk 2]
```diff
@@ -103,122 +110,110 @@ const StyledButton = styled(Attr.button)`
...
-                            </li>}
-                        </ul >
-                    </Attr.div>
-                </ThemeProvider>
-            </MastheadContainer>
-        );
-    }
-}
+                        </li>}
+                        {more && !more.selected && items}
```
Comment (mryanlo) [inline lib/components/Masthead/Masthead.tsx]: this seems weird to me. are we actually saying "show items if more exists yet we didn't select it"? or are we trying to say something else?

Comment (PatoBeltran) [reply lib/components/Masthead/Masthead.tsx]: yeah that it is exactly what it is saying haha... the thing is that the masthead on smaller screens hides the items and shows a chevron so you can open up the items in a `more` popup menu.... so this logic is there to make that hidding possible

File: lib/components/Masthead/Masthead.module.scss [hunk 3]
```diff
@@ -186,139 +118,140 @@ $search-max-size: 100 * $grid-size;
...
-    &.masthead-toolbar-btn {
-      &:hover {
-        background-color: var(--color-bg-masthead-hover);
-      }
-
-      &.selected {
-        background-color: var(--color-bg-container-panel) !important;
-        color: var(--color-text-rest) !important;
+    &.selected {
+      background-color: var(--color-bg-container-panel) !important;
```
Comment (msdehoward) [inline lib/components/Masthead/Masthead.module.scss]: is there a way around these important tags?

File: lib/components/Masthead/Masthead.tsx [hunk 4]
```diff
@@ -103,122 +110,110 @@ const StyledButton = styled(Attr.button)`
...
+                                    icon='chevronRight'
+                                    size={IconSize.xsmall}
+                                    className={cx({
+                                    'nav-icon-collapsed': !navigation.isExpanded,
+                                    'nav-icon-expanded': navigation.isExpanded, 
+                                })} />
+                        </StyledButton>
+                    <InlinePopup.Panel alignment='left' className={cx('nav-panel')}>
+                        {navigation.children}
+                        { navigation.farBottomChildren &&
```
Comment (msdehoward) [inline lib/components/Masthead/Masthead.tsx]: nit: extra space here & on line 161

**Pull request merged**